Command/Proxy/PingDoctrineCommand.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Command\Proxy;
+
+use Doctrine\DBAL\Tools\Console\Command\PingCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Ping the database to check its accessible.
+ */
+class PingDoctrineCommand extends PingCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $defaultName = 'doctrine:ping';
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        $this
+            ->addOption('connection', null, InputOption::VALUE_OPTIONAL, 'The connection to use for this command');
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        DoctrineCommandHelper::setApplicationConnection($this->getApplication(), $input->getOption('connection'));
```

Comment:
I think it would be possible to use [a configurator](https://symfony.com/doc/current/service_container/configurators.html) to achieve the `addOption` call, but I can't think of a way to have the functional equivalent of this line without resorting to a console listener and exposing the input… so I guess extension is the only way here, unless someone has an idea…